backend/mobilityAI/Controllers/PatientsController.cs

```diff
@@ -0,0 +1,180 @@
+            if(data.Count != 0) {
+
+                float[] count = new float[5];
+                float[] total = new float[5];
+                float[][] activityTotals = new float[5][];
+                for (int i = 0; i < 5; i++) activityTotals[i] = new float[13];
+                float totalRows = data.Count;
+
+                for(int i = 0; i < data.Count-1; i++) {
+                    var element = data[i];
+                    var nextElement = data[i+1];
+
+                    count[element.Type]++;
+
+                    int startHour = (FromUnixTime(element.Start/1000).Hour) - 12;
+                    activityTotals[element.Type][startHour] += (nextElement.Start - element.Start)/1000f;
+                }
+
+                int startHour2 = (FromUnixTime(data[data.Count-1].Start).Hour) - 12;
+                activityTotals[data[data.Count-1].Type][startHour2] += (data[data.Count-1].End - data[data.Count-1].Start)/1000f;
+
+                for (int i=0; i<5; i++){
+                    for (int j=0; j<13; j++){
+                        activityTotals[i][j] = activityTotals[i][j]/60;
+                    }
+                }
+
+                total[(int)ActivityType.sitting] = (count[(int)ActivityType.sitting] / totalRows) * 100;
+                total[(int)ActivityType.lyingDown] = (count[(int)ActivityType.lyingDown] / totalRows) * 100;
+                total[(int)ActivityType.walking] = (count[(int)ActivityType.walking] / totalRows) * 100;
+                total[(int)ActivityType.standing] = (count[(int)ActivityType.standing] / totalRows) * 100;
+                total[(int)ActivityType.unknown] = (count[(int)ActivityType.unknown] / totalRows) * 100;
+
+                var retObj = new
+                {
+                    sitting = new
+                    {   
+                        total = total[(int)ActivityType.sitting],
+                        bar = activityTotals[(int)ActivityType.sitting]
+                    },
+                    lyingDown = new
+                    {
+                        total = total[(int)ActivityType.lyingDown],
+                        bar = activityTotals[(int)ActivityType.lyingDown]
+                    },
+                    walking = new
+                    {
+                        total = total[(int)ActivityType.walking],
+                        bar = activityTotals[(int)ActivityType.walking]
+                    },
+                    standing = new
+                    {
+                        total = total[(int)ActivityType.standing],
+                        bar = activityTotals[(int)ActivityType.standing]
+                    },
+                    unknown = new
+                    {
+                        total = total[(int)ActivityType.unknown],
+                        bar = activityTotals[(int)ActivityType.unknown]
+                    },
+                };
+
+
+                return Ok(JsonConvert.SerializeObject(retObj));
+            }
+
+            return Ok();
```

Comment:
Should this return some kind of error code? (We can fix that for next time, just wondering)

Reply:
There's a lot of things we need to fix in this endpoint 😂
Just doing the api reorganization stuff for now